library/src/include/tensile_host.hpp

```diff
@@ -92,15 +103,22 @@ struct RocblasContractionProblem
         , k(k)
         , alpha(alpha)
         , A(A)
-        , ld_a(ld_a)
-        , stride_a(stride_a)
+        , row_stride_a(1)
+        , col_stride_a(ld_a)
+        , batch_stride_a(batch_stride_a)
```

Comment:
I would have thought you would need:
if (trans_a == rocblas_operation_none)
{
    row_stride_a == 1;
    col_stride_a  == lda;
}
else
{
    row_stride_a = lda;
    col_stride_a = 1;
}

is there some other place where the row and col strides are switched to account for transposition of A ?

Reply:
> is there some other place where the row and col strides are switched to account for transposition of A ?

Yes, [in the new Tensile client](https://github.com/leekillough/rocBLAS/blob/0878f28525ff914b686fcb69786098c49af0acbc/library/src/tensile_host.cpp#L138).

`tensile_host.hpp` only _sets up_ problems; they are _solved_ in `tensile_host.cpp`.

This PR is still being heavily edited; it should be considered a draft right now.